src/applicationMicroservices/bucketService/src/main/java/com/shop/bucketservice/controller/BucketController.java

```diff
@@ -4,39 +4,63 @@
 import com.shop.bucketservice.service.BucketService;
 import com.shop.bucketservice.service.impl.BucketServiceImpl;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
 
 @RestController
-@RequestMapping("/buckets")
+@RequestMapping("/api/v1/buckets")
 public class BucketController {
 
     @Autowired
     BucketService bucketService = new BucketServiceImpl();
 
     @GetMapping
-    public List<Bucket> getAllBuckets() {
-        return bucketService.getAllBuckets();
+    public ResponseEntity<List<Bucket>> getAllBuckets(@RequestHeader String role) {
+        if(role.equals("USER")){
+            return ResponseEntity.status(404).body(null);
+        }
+        return ResponseEntity.ok(bucketService.getAllBuckets());
     }
 
     @PostMapping("/add")
-    public void addProductToBucket(@RequestBody Bucket bucket) {
-        bucketService.addProductToTheBucket(bucket);
+    public ResponseEntity<String> addProductToBucket(@RequestBody Bucket bucket, @RequestHeader String role) {
+        if(role.equals("USER")){
+            bucketService.addProductToTheBucket(bucket);
+            return ResponseEntity.ok("done");
+        } else {
+            return ResponseEntity.status(404).body("error");
+        }
     }
 
     @DeleteMapping("/delete")
-    public void deleteProductFromBucket(@RequestBody Bucket bucket) {
-        bucketService.deleteProductFromTheBucket(bucket);
+    public ResponseEntity<String> deleteProductFromBucket(@RequestBody Bucket bucket, @RequestHeader String role) {
+        if(role.equals("USER")){
+            bucketService.deleteProductFromTheBucket(bucket);
+            return ResponseEntity.ok("done");
+        } else {
+            return ResponseEntity.status(404).body("error");
+        }
     }
 
     @DeleteMapping("/clear")
-    public void clearBucket(@RequestParam int bucketId) {
-        bucketService.clearBucket(bucketId);
+    public ResponseEntity<String> clearBucket(@RequestParam int bucketId, @RequestHeader String role) {
+        if(role.equals("USER")){
+            bucketService.clearBucket(bucketId);
+            return ResponseEntity.ok("done");
+        } else {
+            return ResponseEntity.status(404).body("error");
+        }
     }
 
     @PostMapping("/makeOrder")
```

Comment:
dont use verbs 
https://www.partech.nl/nl/publicaties/2020/07/9-trending-best-practices-for-rest-api-development

Reply:
done